### PR TITLE
Add use_firehose option & configuration conflit

### PR DESCRIPTION
**Pull request by dopch**

### CODE REVIEW

**Comment by jordansissel** [comment]
jenkins, test it

File: lib/logstash/inputs/twitter.rb [hunk 1]
```diff
@@ -101,10 +104,14 @@ class LogStash::Inputs::Twitter < LogStash::Inputs::Base
   def register
     require "twitter"
 
-    if !@use_samples && ( @keywords.nil? && @follows.nil? && @locations.nil? )
+    if !@use_samples && !@use_firehose && ( @keywords.nil? && @follows.nil? && @locations.nil? )
       raise LogStash::ConfigurationError.new("At least one parameter (follows, locations or keywords) must be specified.")
     end
 
+    if @use_samples && @use_firehose
+      raise LogStash::ConfigurationError.new("Only use_firehose or use_samples can be choose")
```
Comment (jordansissel) [inline lib/logstash/inputs/twitter.rb]: Typo on `choose` - maybe rephrase "use_firehose and use_samples cannot both be true. Choose one or the other." ?

**Comment by jordansissel** [comment]
Code change looks OK to me. I don't have access to the twitter firehose so I'm not sure how to test this new change.

I left one small comment on the validation error. Back to you to respond - otherwise this change looks good to me! :)

**Comment by dopch** [comment]
you can try the method Twitter firehose send your a reply "Unauthorized" , I change the validation error =)
